aws_google_auth/google.py

```diff
@@ -206,8 +209,72 @@ def do_login(self):
         self.check_extra_step(response_page)
 
         if cap is not None:
-            raise ExpectedGoogleException(
-                'Captcha Required. Manually Login to remove this.')
+            self.session.headers['Referer'] = sess.url
+
+            # Collect ProfileInformation, SessionState, signIn, and Password Challenge URL
+            challenge_page = response_page
+
+            profile_information = challenge_page.find('input', {
+                'name': 'ProfileInformation'
+            }).get('value')
+            session_state = challenge_page.find('input', {
+                'name': 'SessionState'
+            }).get('value')
+            sign_in = challenge_page.find('input', {'name': 'signIn'}).get('value')
+            passwd_challenge_url = challenge_page.find('form', {
+                'id': 'gaia_loginform'
+            }).get('action')
+
+            # Update the payload
+            payload['SessionState'] = session_state
+            payload['ProfileInformation'] = profile_information
+            payload['signIn'] = sign_in
+            payload['Passwd'] = self.config.password
+
+            # Get all captcha challenge tokens and urls
+            captcha_container = challenge_page.find('div', {'class': 'captcha-container'})
+            captcha_logintoken = captcha_container.find('input', {'name': 'logintoken'}).get('value')
+            captcha_url = captcha_container.find('input', {'name': 'url'}).get('value')
+            captcha_logintoken_audio = captcha_container.find('input', {'name': 'logintoken_audio'}).get('value')
+            captcha_url_audio = captcha_container.find('input', {'name': 'url_audio'}).get('value')
+
+            # Open captcha image
+            url = urlopen(captcha_url)
```

Comment:
Could we rewrite some of this with the `with X as Y:` style? Like this:

```python
with urlopen(captcha_url) as url:
    with io.BytesIO(url.read()) as f:
        img = Image.open(f)
img.show()
```

This handles closing the file handles automatically, and covers the cases where there are exceptions.